### ISSUE TITLE
[Bug]: No completion items for record fields of record typed named argument 

### ISSUE THREAD

**Issue by malinthar** [ballerina-platform/ballerina-lang#39296]
### Description

![image](https://user-images.githubusercontent.com/35211477/213246679-ad50168c-89d7-4b37-88b0-0b17634b3625.png)

The `secureSocket` argument is of the following type. 
```Ballerina
public type ListenerSecureSocket record {|
    crypto:KeyStore|CertKey key;
    record {|
        VerifyClient verifyClient = REQUIRE;
        crypto:TrustStore|string cert;
    |} mutualSsl?;
    record {|
        Protocol name;
        string[] versions = [];
    |} protocol?;
    record {|
        CertValidationType 'type = OCSP_STAPLING;
        int cacheSize;
        int cacheValidityPeriod;
    |} certValidation?;
    string[] ciphers = ["TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256", "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
                        "TLS_DHE_RSA_WITH_AES_128_CBC_SHA256", "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
                        "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA", "TLS_DHE_RSA_WITH_AES_128_CBC_SHA",
                        "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256", "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
                        "TLS_DHE_RSA_WITH_AES_128_GCM_SHA256"];
    boolean shareSession = true;
    decimal handshakeTimeout?;
    decimal sessionTimeout?;
|};

```

The fields are not suggested as completion items. 

### Steps to Reproduce

Mentioned above

### Affected Version(s)

2201.3.2

### Related area

-> Editor

**Comment by malinthar** [comment]
With https://github.com/ballerina-platform/ballerina-lang/pull/38596 we have introduced the expected type API and we have used it in the language server for obtaining the context type for a given cursor position. Using the expected type API has fixed the above issue. 
![image](https://user-images.githubusercontent.com/35211477/213982837-6dec11b6-07b4-4355-8aca-e636d6dfeed0.png)

However, the type provided by the expected type API for the following scenario is not accurate. Therefore, the completion items provided are not correct as well.

![image](https://user-images.githubusercontent.com/35211477/213983370-042219c1-b42f-458e-82eb-03f6ed8ecff3.png)

@DamithaSenevirathne

```Ballerina
import ballerina/http;

listener http:Listener securedEp = new (9090, secureSocket = {'key: {<cursor>}});
```

**Comment by IMS94** [comment]
@malinthar that's great. Let's mark this as blocked by #38596 for now.

**Comment by malinthar** [comment]
Blocked by https://github.com/ballerina-platform/ballerina-lang/pull/38596

**Comment by DamithaSenevirathne** [comment]
I will be working on a fix related to this (https://github.com/ballerina-platform/ballerina-lang/pull/38596)

**Comment by IMS94** [comment]
https://github.com/ballerina-platform/ballerina-lang/pull/38596 was merged. We can verify and close this issue

**Comment by DamithaSenevirathne** [comment]
> I will be working on a fix related to this (#38596)

Created a separate issue to track this.
